src/c_promenne.md

```diff
@@ -1,53 +1,119 @@
-požadovanou hodnotu. Typicky se toto děje, pokud používáme proměnnou
-jako nějaké počítadlo.
-
+Proměnná je platná (lze ji používat) vždy od místa
+deklarace do konce **bloku**, ve kterém byla nadeklarována. Bloky jsou kusy kódu ohraničené složenými
+závorkami (`{` a `}`):
 ```c
-int counter;
+int main() {
+    int a;
+
+    {
+        // zde je platné pouze `a`
+        int b;
+        // zde je platné `a` i `b`
+    } // zde končí platnost proměnné `b`
 
-int sum, no_elements = 0;
+    // zde je platné pouze `a`
 
-printf( "sum: %d\n", sum );                  // muze byt cokoli: sum: 432749
-printf( "no_elements: %d\n", no_elements );  // no_elements: 0
+    return 0;
+} // zde končí platnost proměnné `a`
 ```
 
-Na řádku 5 a 6 vidíme použití knihovní funkce `printf`, která nám bude sloužit pro
-tisk obsahu proměnných na konzoli. Můžete vidět, že funkce `printf` používá
-formátovací řetězec, kde celá čísla (`int`eger) jsou zastoupena znakem `%d`.
-Obecně je formátovácí řetězec uvozen znakem `%` následovaný formátovacím
-znakem. Více o formátování výstupu se můžete dozvědet např. zde:
-[^1],[^2].
+### Datový typ
+`int` před názvem proměnné udává její datový typ, o kterém pojednává [následující sekce](c_datove_typy.md).
+Prozatím si řekněme, že `int` je zkratka pro `integer`, tedy celé číslo. Tím říkáme programu, že má
+tuto proměnnou (resp. paměť, kterou proměnná reprezentuje) interpretovat jako číslo.
+
+### Inicializace
+Do proměnné můžeme při jejím vytvoření rovnou uložit nějaký *výraz*, který musí být stejného datového
+typu jako je typ proměnné:
+```c
+int points_a = 10;
+int points_b = 10 + 15;
+```
+Obecná syntaxe pro deklaraci proměnné je
+
+`<datový typ> <název>;`
+
+popřípadě
 
+`<datový typ> <název> = <výraz>;`
+
+pokud použijeme inicializaci.
+
+> Všimněte si, že na konci deklarace proměnné vždy musí následovat středník (**;**).
+> Opomenutí středníku na konci příkazu je velmi častá chyba, která často končí těžko srozumitelnými chybovými
+> hláškami při překladu. Dávejte si tak na středníky pozor, obzvláště ze začátku.
+
+### Čtení
+Pokud v programu použijeme název platné proměnné, tak dojde k načtneí její hodnoty.
+Pokud použijeme název proměnné v programu na místě, kde je očekáván výraz, tak se vyhodnotí jako
+současná hodnota proměnné:
 ```c
-int a = 5, b = 10;
+int main() {
+    int a = 5;
+    int b = a;  // hodnota `b` je 5
+    int c = b + a + 1;  // hodnota `c` je 11
+}
+```
 
-int c = a + b;
+Kdekoliv tak můžete použít výraz, můžete použít i proměnnou (pokud sedí datové typy). Pro výpis hodnot
+proměnných na výstup programu můžete `printf`. Hodnoty proměnných můžete zkoumat také krokováním
+pomocí [debuggeru](ladeni.md#krokování).
 
-printf( "%d + %d = %d\n", a, b, c );  // 5 + 10 = 15
+### Zápis
+Pokud by proměnná měla pouze svou původní hodnotu, tak by nebyla moc užitečná. Hodnoty proměnných
+naštěstí jde měnit. Můžeme k tomu použít další typ C příkazu, tzv. přiřazení (**assignment**):
+```c,editable,readonly
```

Comment:
k cemu to spusteni je? :) nepridat tam printf?

Reply:
Jasně, tady není k ničemu, dal jsem pryč, díky.